tests/Offer/Popularity/PopularityTest.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Offer\Popularity;
+
+use InvalidArgumentException;
+use PHPUnit\Framework\TestCase;
+use TypeError;
+
+class PopularityTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function it_stores_an_int_value(): void
+    {
+        $popularity = new Popularity(123);
+
+        $this->assertEquals(123, $popularity->toNative());
+    }
+
+    /**
+     * @test
+     */
+    public function it_throws_on_invalid_values(): void
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Popularity can\'t be smaller than zero.');
+
+        new Popularity(-1);
+    }
+
+    /**
+     * @test
+     * @dataProvider inValidTypeProvider
+     * @param mixed $value
+     */
+    public function it_throws_on_invalid_types($value): void
+    {
+        $this->expectException(TypeError::class);
```

Comment:
Nit: This is of course only the case in this test file because it has `declare(strict_types=1);`
If you'd call `new Popularity()` with a float, string, etc from another place where there are no `strict_types` used it won't throw. Instead, it will automatically cast. But there's not much we can do about that. Maybe a small change we can do is change the test name to `it_throws_on_invalid_types_if_strict_types_is_enabled`

Reply:
Another option could be to add an extra validation:
```
if (!is_int($value)) {
    throw new TypeError('Popularity should be an integer.');
}
```

Reply:
That wouldn't do anything unless you remove the type hint in the constructor. Because the type hint in the constructor makes it cast automatically to `int` if it's called without `strict_types`.

Reply:
But I think we should leave it like this (or at most change the test name) and move towards using `strict_types` consistently in the long run.

Reply:
Ah yes, the typehint and implicit typecasting. I'm getting so used to `strict_types` 😀